syntax = "proto3";

package ranking;

service RankingService {
  rpc RankPOIs(RankingRequest) returns (RankingResponse);
}

message RankingRequest {
  repeated float user_embedding = 1;
  string social_mode = 2;
  string intensity = 3;
  int32 top_k = 4;
  repeated string categories_filter = 5;
  optional int32 start_time_minutes = 6;
}

message RankingResponse {
  repeated ScoredPOI scored_pois = 1;
}

message ScoredPOI {
  int32 poi_id = 1;
  string name = 2;
  double lat = 3;
  double lon = 4;
  string category = 5;
  repeated string tags = 6;
  string description = 7;
  int32 avg_visit_minutes = 8;
  double rating = 9;
  double score = 10;
  repeated float embedding = 11;
  string social_mode = 12;
  string intensity_level = 13;
  string open_time = 14;
  string close_time = 15;
}
